### DISCUSSION TITLE
Including nested records

### DISCUSSION THREAD

**Discussion by daxidngyn** [prisma/prisma#12292] [Q&A] [answered]
Is it possible to include nested relations?

From the docs, this example was shown:
```
const getAuthor = await prisma.user.findUnique({
  where: {
    id: "20",
  },
  include: {
    posts: true, // All posts where authorId == 20
  },
});
```

If I had another relation model with posts, such as comments, how would I include the comments as well? Referring to this example, I'm currently having to do to a findUnique() call to get the author, and then another findMany() call to get every comment associated with the author's post. Is there a better way of doing this?

**Comment by austincrim** [comment, marked as answer]
Hey @daxidngyn 👋 ,

You can arbitrarily include as many nested relations as you need, using similiar syntax to what you referenced above. If your `posts` have a `comments` relation, it would look like this:

```ts
const getAuthor = await prisma.user.findUnique({
  where: { id: 20 },
  include: {
    posts: {
      select: {
        comments: true,
        // any other fields you need
      }
    }
  }
})
```

You can read more about this functionality [here in our docs](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#select-specific-relation-fields).

Let me know if I can clarify anything further!

**Reply by Criedor** [reply]
hey @austincrim ,
is there a way to include all nested relations without having to reference them all one by one? Something like:
```
const getAuthor = await prisma.user.findUnique({
  where: { id: 20 },
     include: {
        all: true
     }
  }
})
```

**Reply by austincrim** [reply]
Hello again 👋 ,

I don't think that is possible today, but I can definitely see the utility of such an option. We have a similar feature request [here](https://github.com/prisma/prisma/issues/9777), but feel free to add to that one or [create a new one](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) if you feel your use case wouldn't be covered by it.

In the meantime, you could create a utility function that fetches a `user` and all relations, so you at least don't have to duplicate the query.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
